plugins/uf2/fu-uf2-device.c

```diff
@@ -157,6 +157,11 @@ fu_uf2_device_write_firmware(FuDevice *device,
 		return FALSE;
 	}
 
+	if (fu_device_has_flag(device, FWUPD_DEVICE_FLAG_WILL_DISAPPEAR)) {
```

Comment:
does this get copied from the runtime to the uf2 device? I expected to need `FuDeviceClass->replace()` and check and then set...

Reply:
fu_device_list_replace would have copied it, but that made things worse as it would get lost from the bootloader device. I've changed fu_device_list_replace to not remove the flag if it's already there on the bootloader.